### ISSUE TITLE
Getting a better understanding of `make_row_groups()` and `write_column()`

### ISSUE THREAD

**Issue by yohplala** [dask/fastparquet#695]
Hi,
These are general questions regarding `writer.make_row_groups()` and `writer.write_column()`.
Please, if not appropriate, do not hesitate to advise me a better place for these kind of questions.

**1. `writer.make_row_groups()`**
I find the docstring not so clear. After reading the function, could we consider following docstring instead:
_Write data to Parquet file as single row group, returning row group metadata._
or
_Write data to Parquet file as single row group, returning row group thrift structure._
(not so sure what is thrift about, so not sure about the correct wording)
?

 **2. `writer.write_column()`**
I can see 3 calls to `compress_data()`:

- in a 1st `if` branch ` if is_categorical_dtype(data.dtype):`
```python
        bdata = encode['PLAIN'](pd.Series(data.cat.categories), selement)
        l0 = len(bdata)
        if compression and compression.upper() != "UNCOMPRESSED":
            bdata = compress_data(bdata, compression)
            l1 = len(bdata)
        else:
            l1 = l0
```

- in a 2nd `if` branch `if datapage_version == 1:`
```python
        bdata = b"".join([
            repetition_data, definition_data, encode[encoding](data, selement), 8 * b'\x00'
        ])
        [...]
        if compression:
            bdata = compress_data(bdata, compression)
            l1 = len(bdata)
        else:
            l1 = l0
```

- in a 3rd `if` branch `elif datapage_version == 2:`
```python
        bdata = encode[encoding](data, selement)
        lb = len(bdata)
        if is_compressed:
            bdata = compress_data(bdata, compression)
            diff = lb - len(bdata)
        else:
            diff = 0
```

My questions may be a bit vague.

- Basically, I don't have a clear understanding about what is inside `bdata`?
    - metadata + data
    - or data only?
- Why are categoricals processed in a 1st `if` branch?
  Is not `data` compressed a 2nd time when being managed in one of the subsequent `if` branch?
  I am aware this question sounds very stupid, but I fail to see what prevents `data` to be compressed 2 times, a 1st time in the 1st `if` branch, and a 2nd time in one of the 2 following `if` branch.

Thanks in advance for your feedbacks!
Bests
